app/src/main/java/cloudcity/dataholders/MetricsPOJO.java

```diff
@@ -0,0 +1,124 @@
+package cloudcity.dataholders;
+
+/**
+ * Plain Old Java Object (POJO) for holding iPerf3 network performance metrics.
+ * Contains statistical data for both download (DL) and upload (UL) measurements.
+ *
+ * Used for sending data to CC server.
+ */
+public class MetricsPOJO {
+
+    private final double DLmin;
+    private final double DLmedian;
+    private final double DLmean;
+    private final double DLmax;
+    private final double DLlast;
+
+    private final double ULmin;
+    private final double ULmedian;
+    private final double ULmean;
+    private final double ULmax;
+    private final double ULlast;
+
+    /**
+     * The all arg constructor. Please prefer using the other {@link #MetricsPOJO(DownloadMetrics, UploadMetrics)}
+     * over this one, as this one might be error-prone
+     *
+     * @param DLmin download min
+     * @param DLmedian download median (average) of the whole sampling list
+     * @param DLmean download mean (middle element of sampling list)
+     * @param DLmax download max
+     * @param DLlast last download sample
+     * @param ULmin upload min
+     * @param ULmedian upload median (average) of the whole sampling list
+     * @param ULmean upload mean (middle element of the sampling list)
+     * @param ULmax upload max
+     * @param ULlast last upload sample
+     */
+    public MetricsPOJO(double DLmin, double DLmedian, double DLmean, double DLmax, double DLlast,
+                       double ULmin, double ULmedian, double ULmean, double ULmax, double ULlast) {
+        this.DLmin = DLmin;
+        this.DLmedian = DLmedian;
+        this.DLmean = DLmean;
+        this.DLmax = DLmax;
+        this.DLlast = DLlast;
+        this.ULmin = ULmin;
+        this.ULmedian = ULmedian;
+        this.ULmean = ULmean;
+        this.ULmax = ULmax;
+        this.ULlast = ULlast;
+    }
```

Comment:
ah, but I looked at the implementation and that's what those two are. I looked because I was unsure why they're using both "mean" and "median" when they should be the same thing; however, they're not. mean is "average", and median is literally "the (lastIndex/2)-th element".

thank you for the suggestion but unfortunatelly, it's wrong, and you could have read the code as well. 
please learn to **contextually** read code you're about to suggest improvements on. meaning, read where it's used and what's it used with, and not just nitpicking on "what it should be and hence what it usually is" when that might not be the case.

resolving